### PR TITLE
Wingweight

### PR DESCRIPTION
**Pull request by m-balesdent**
Changes of wingweight example to answer  @regislebrun remarks

### CODE REVIEW

**Comment by jschueller** [comment]
dont you want to include the changes from #2148 here ?

**Comment by m-balesdent** [comment]
> seems the example is way too slow now and exceeds the timeout (100s) <10s is preferred

The example is a tutorial to compare a lot of sensitivity analysis methods (SRC, SRRC, Taylor, Pearson, Sobol with Saltelli and Sobol with polynomial chaos, HSIC). So it can take much time than classical example ;)

Before, the polynomial chaos was done in "expert mode" in which all the basis, truncation, input distribution were defined by the user. To follow Regis' remark, this part has been changed to a "push-button" mode in which only the input and output samples are given. Consequently, the chaos polynomial algorithm has to estimate the distribution, basis, etc. that may cause this increase of duration.

**Comment by jschueller** [comment]
If you have the distribution you can still pass it to the simplified constructor.

**Comment by jschueller** [comment]
it's the hsic computation at the end that take forever, try lowering the doe size

**Comment by m-balesdent** [comment]
> it's the hsic computation at the end that take forever, try lowering the doe size

Thanks. I've changed that point and it now takes 50s on my laptop.

**Comment by jschueller** [comment]
50s is still a bit high, can you reduce again ? let's say at most 20s

**Comment by m-balesdent** [comment]
> 50s is still a bit high, can you reduce again ? let's say at most 20s

Now it takes 17s. Is it OK for you?

**Comment by jschueller** [comment]
great, thanks

File: python/doc/examples/reliability_sensitivity/sensitivity_analysis/plot_sensitivity_wingweight.py [hunk 1]
```diff
@@ -370,17 +360,19 @@
 # %%
 #
 # The Sobol' indices confirm the previous analyses, in terms of ranking of the most influent variables. We also see that five variables have a quasi null total Sobol' indices, that indicates almost no influence on the wing weight.
-# There is no discrepancy between first order and total Sobol' indices, that indicates no or very low interaction between the variables in the variance of the output.
+# There is no discrepancy between first order and total Sobol' indices, that indicates no or very low interaction between the variables in the variance of the output. 
+# As the most important variables act only through decoupled first degree contributions, the hypothesis of a linear dependence between the input variables and the weight is legitimate, which explains why both squared SRC and Taylor give the exact same results even if the first one is based on a :math:`\mathcal{L}^2` linear approximation and the second one is based on a linear expansion around the mean value of the input variables.
```
Comment (jschueller) [inline python/doc/examples/reliability_sensitivity/sensitivity_analysis/plot_sensitivity_wingweight.py:364]: these lines are way too long and make the review difficult

**Review by jschueller — APPROVED** [review]

File: python/doc/examples/reliability_sensitivity/sensitivity_analysis/plot_sensitivity_wingweight.py (hunk 1 above)
Comment (m-balesdent) [reply python/doc/examples/reliability_sensitivity/sensitivity_analysis/plot_sensitivity_wingweight.py]: > these lines are way too long and make the review difficult

The line has been splitted.

**Review by sofianehaddad — APPROVED** [review]

**Pull request merged**